tests/test_operation.py

```diff
@@ -2554,19 +2555,13 @@ def test_has_grad_method(self):
         assert qml.operation.has_grad_method(self.rot)
         assert not qml.operation.has_grad_method(self.cnot)
 
-    @pytest.mark.usefixtures("use_legacy_opmath")
-    def test_gen_is_multi_term_hamiltonian_legacy_opmath(self):
+    def test_gen_is_multi_term_hamiltonian(self):
```

Comment:
The `gen_is_multi_term_hamiltonian` function is used in decomposition/tape expansion logic, and is not just a type check. This function should not have been updated to run with `use_legacy_opmath`; its failure indicates a genuine problem (related to the QNG test failure in optimize), and not just a difference in convention.

Reply:
Yes, it has been modified recently with temporary fixes to enable opmath globally, so I'm not surprised that we missed this